pkg/apiserver/apiserver.go

```diff
@@ -38,8 +38,23 @@ import (
 
 	"github.com/emicklei/go-restful"
 	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
+var (
+	apiserverLatencies = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "apiserver_request_latencies",
+			Help: "Response latency summary in microseconds for each request handler, verb, and HTTP response code.",
+		},
+		[]string{"handler", "verb", "code"},
```

Comment:
What do you envision being the handlers? Today I see `redirect` and `rest`. I'm not sure what information you're interested in gathering, would breaking down by resource be more useful?